### ISSUE TITLE
[BUG] Add-PnPApp fails with a ResourceNotFound exception.

### ISSUE THREAD

**Issue by RobinDijkstra** [pnp/powershell#1798]
Add-PnPApp fails with a ResourceNotFound exception.

After creating a new team site and creating the site app catalog, it's not possible to add an app package. When checking the site in Sharepoint, the app catalog does exists. Some other checks:

- Get-PnPApp -Scope Site : same error
- Get-PnPList -Identity AppCatalog : no problem
- Get-PnPSiteCollectionAppCatalogs : Returns ErrorMessage = "Access denied." for the created site.

### Reporting an Issue or Missing Feature
Issue

### Expected behavior 
App should be uploaded into the site app catalog

### Actual behavior 
![image](https://user-images.githubusercontent.com/38421680/164725219-4f3490c1-a46b-4448-9a7c-7e03644bdfe1.png)

### Steps to reproduce behavior
Connect-PnPOnline -Url $adminUrl -Interactive
$newSiteUrl = New-PnPSite -Type TeamSite -Title $siteAlias -Alias $siteAlias
Add-PnPSiteCollectionAppCatalog -Site $newSiteUrl
Connect-PnPOnline -Url $newSiteUrl -Interactive
Add-PnPApp -Path xyz.sppkg -Scope Site

### What is the version of the Cmdlet module you are running?
1.10.0

### Which operating system/environment are you running PnP PowerShell on?
- [ X ] Windows
- [ ] Linux
- [ ] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by gautamdsheth** [comment]
@RobinDijkstra - did you create a site collection app catalog ? 

You can use `Add-PnPSiteCollectionAppCatalog` to first create the app catalog and then upload your apps 

https://pnp.github.io/powershell/cmdlets/Add-PnPSiteCollectionAppCatalog.html

**Comment by RobinDijkstra** [comment]
@gautamdsheth - Yes, I did. See step 3 in the steps to reproduce. I've also checked that the site collection app catalog is actually present.

**Comment by gautamdsheth** [comment]
Can you maybe try this in another site collection and check or maybe in a different tenant ? Could be some site collection or tenant specific issue ? Not able to reproduce it.

**Comment by RobinDijkstra** [comment]
I've tried the steps to reproduce several times on our two tenants and also on my personal develop tenant. On all tenants the behavior is the same.
The problem is only with new sites, adding an app to older, already existing sites works just fine.
The steps to reproduce are a simplified version of an extended set of scripts to create and configure new sites. These scripts have always worked and they don't work anymore. So something must have changed.

**Comment by AndersRask** [comment]
Could this be race conditions? 
Depending on what kind of site you create (especially TeamSites that are more complex) the site may not be ready yet to receive your app.
Please for the sake of testing try and add a Start-Sleep 120 and see if the behaviour changes
hth

**Comment by RobinDijkstra** [comment]
After a few hours I've tried the Add-PnpApp cmdlet again, it made no difference. I've also tried it on a site I've created a few days ago and that also didn't work.

**Comment by RobinDijkstra** [comment]
I've found the problem. For those of you who run into the same issue (and those who are just interested):

According to [MS documentation](https://docs.microsoft.com/en-us/sharepoint/dev/general-development/site-collection-app-catalog#create-a-site-collection-app-catalog), the account that is used to create the site collection app catalog must be Site Collection Administrators on both the tenant-level App Catalog and the target Site Collection. Normally I can ignore these kinds of remarks because I'm a Global Administrator or a Sharepoint Administrator. In this case however it's really necessary to be a Site Collection Administrator on both sites.
So I've added myself as a Site Collection Administrator to the tenant-level App Catalog site. In the steps to reproduce I've added a step to be Site Collection Administrator in the newly created site, before creating the site collection app catalog:

Connect-PnPOnline -Url $adminUrl -Interactive
$newSiteUrl = New-PnPSite -Type TeamSite -Title $siteAlias -Alias $siteAlias
**_Set-PnPtenantSite -Identity $newSiteUrl -Owners $loginName_**
Add-PnPSiteCollectionAppCatalog -Site $newSiteUrl
Connect-PnPOnline -Url $newSiteUrl -Interactive
Add-PnPApp -Path xyz.sppkg -Scope Site

This is the solution for all my three tenants. Still, I don't know what has changed because it used to work. I suspect it has something to do with the [change](https://admin.microsoft.com/?ref=MessageCenter/:/messages/MC329484) that MS has rolled out on the App Catalog Site (which now is called the Apps site).

@gautamdsheth, @AndersRask: thanks for your input!
